src/subdomains/supporting/bank-tx/bank-tx/services/bank-tx.service.ts

```diff
@@ -154,6 +155,68 @@ export class BankTxService {
     }
   }
 
+  async fillBankTx(): Promise<void> {
+    const entities = await this.bankTxRepo.find({
+      where: {
+        accountingAmountBeforeFee: IsNull(),
+        amount: Not(IsNull()),
+        chargeAmount: Not(IsNull()),
+      },
+      relations: { buyCrypto: true, buyFiats: true },
+    });
+
+    for (const entity of entities) {
+      try {
+        if (![BankTxType.BUY_CRYPTO, BankTxType.BUY_FIAT].includes(entity.type)) {
+          await this.bankTxRepo.update(entity.id, {
+            accountingAmountBeforeFee: Util.roundReadable(entity.amount + entity.chargeAmount, AmountType.FIAT),
+          });
+          continue;
+        }
+
+        const accountingFeePercent = entity.buyCrypto?.percentFee ?? entity.buyFiats?.[0]?.percentFee;
+        const accountingFeeAmount =
+          accountingFeePercent *
+          (entity.type === BankTxType.BUY_CRYPTO
+            ? entity.amount + entity.chargeAmount
+            : entity.type === BankTxType.BUY_FIAT
+            ? (entity.amount + entity.chargeAmount) / (1 - accountingFeePercent)
+            : undefined);
+        const accountingAmountAfterFee =
+          entity.type === BankTxType.BUY_CRYPTO
+            ? entity.amount + entity.chargeAmount - accountingFeeAmount
+            : entity.type === BankTxType.BUY_FIAT
+            ? entity.amount + entity.chargeAmount
+            : undefined;
+
+        const accountingAmountBeforeFeeChf =
+          entity.type === BankTxType.BUY_CRYPTO
+            ? entity.buyCrypto.amountInChf
+            : entity.type === BankTxType.BUY_FIAT
+            ? entity.buyFiats[0].amountInChf / (1 - accountingFeePercent)
+            : undefined;
+
+        const accountingAmountAfterFeeChf =
+          entity.type === BankTxType.BUY_CRYPTO
+            ? entity.buyCrypto.amountInChf * (1 - accountingFeePercent)
+            : entity.type === BankTxType.BUY_FIAT
+            ? entity.buyFiats[0].amountInChf
+            : undefined;
```

Comment:
Maybe create a single if-else (with entity.type === BankTxType.BUY_CRYPTO) and assign all values together?